src/ldp/alg/tree_search.py

```diff
@@ -91,8 +90,8 @@ async def sample_tree(self, env: TEnv, max_depth: int | None) -> TransitionTree:
         except CaughtError:
             return tree
         finally:
-            # Tear down env resources if necessary
-            await safe_close_env(env, self.catch_env_failures)
```

Comment:
Note I don't think it's desirable to completely suppress away `EnvError` here twice, so I have removed `safe_close_env` usage